src/index.js

```diff
@@ -1,38 +1,219 @@
+import Task from './Task.js';
 import './style.css';
 
 const tasksSection = document.querySelector('.tasks-container');
+const inputBox = document.querySelector('.input-box');
 
-const tasksList = [
-  {
-    task: 'task 1',
-    completed: 'false',
-    index: 0,
-  },
-  {
-    task: 'task 2',
-    completed: 'false',
-    index: 1,
-  },
-  {
-    task: 'task 3',
-    completed: 'false',
-    index: 2,
-  },
-];
-
-function component(task) {
-  const taskElement = document.createElement('div');
-  taskElement.className = 'tasks';
-  taskElement.innerHTML = `
+class ToDoList {
+  constructor() {
+    this.tasksList = [];
+  }
+
+  static getLocalList = () => {
+    const temp = localStorage.getItem('my-to-do-list');
+
+    if (temp) {
+      this.tasksList = JSON.parse(temp);
+    } else {
+      localStorage.setItem('my-to-do-list', JSON.stringify(this.tasksList));
+    }
+
+    return this.tasksList;
+  }
+
+  static updateLocalList = (data) => {
+    localStorage.setItem('my-to-do-list', JSON.stringify(data));
+  }
+
+  static clearAll = () => {
+    // let temp = 0;
+    // temp = this.tasksList.filter((task) => task.completed === false);
+    const completedTasks = this.tasksList.filter((task) => task.completed === true);
+    const allTasks = document.querySelectorAll('.tasks');
+    if (completedTasks.length !== 0) {
+      for (let i = 0; i < allTasks.length; i += 1) {
+        if (this.tasksList[i].completed === true) {
+          document.querySelector('.tasks-container').removeChild(allTasks[i]);
+        }
+      }
+      this.tasksList = this.tasksList.filter((task) => task.completed === false);
+      ToDoList.updateLocalList(this.tasksList);
+    }
+    // re-arrange index after delete
+    if (this.tasksList.length !== 0) {
+      for (let i = 0; i < this.tasksList.length; i += 1) { this.tasksList[i].index = i; }
+    }
+    // update the local storage
+    ToDoList.updateLocalList(this.tasksList);
+  }
+
+  static deleteTask = (target) => {
+    const temp = [];
+    let desc = '';
+    desc = target.parentElement.firstElementChild.lastElementChild.textContent;
+    this.tasksList = ToDoList.getLocalList();
+    console.log(`temp${temp}`);
+
+    this.tasksList.forEach((item, index) => {
+      if ((((item.description).trim().toString()) === (desc.trim().toString()))) {
+        this.tasksList.splice(index, 1);
+        console.log(`after delete ${temp}`);
+        ToDoList.updateLocalList(this.tasksList);
+      }
+    });
+
+    // re-arrange index after delete
+    this.tasksList = ToDoList.getLocalList();
+    if (this.tasksList.length !== 0) {
+      for (let i = 0; i < this.tasksList.length; i += 1) { this.tasksList[i].index = i; }
+    }
+    // update the local storage
+    ToDoList.updateLocalList(this.tasksList);
```

Comment:
- [ ] It would be better if our localStorage works as expected, however, the value being stored in the localStorage now is undefined as shown in the screenshot below
![Screenshot from 2022-08-19 10-17-31](https://user-images.githubusercontent.com/46388113/185599252-62654f37-6295-42ed-beee-c41a51d81079.png)